plugins/inputs/elasticsearch_query/aggregation_parser.go

```diff
@@ -0,0 +1,117 @@
+package elasticsearch_query
+
+import elastic "gopkg.in/olivere/elastic.v5"
```

Comment:
Would it make sense to use the official client at https://github.com/elastic/go-elasticsearch?

Reply:
That would require a complete rewrite of the code I'm afraid.
Also this client is the same already in use on elasticsearch output plugin.

The fact that it is using a rather "old" (v5) version maybe is an issue or will become one soon.
One option would be to import multiple versions (v5, v6, v7...) and use the right client according to the version of the cluster detected.

Reply:
Fair enough. Let's start with this dependency then and maybe switch to the other client with higher versions...